src/mimeparser.rs

```diff
@@ -307,7 +315,6 @@ impl MimeMessage {
                         // && decryption_info.dkim_results.allow_keychange
                         {
                             peerstate.degrade_encryption(message_time);
-                            peerstate.save_to_db(&context.sql).await?;
```

Comment:
Because it may be upgraded later, i.e. to reduce the number of calls to save_to_db(). If `peerstate` is set to Mutual, it's saved immediately to not lose that fact in case of an error. Otherwise it's saved after parsing a mail

Reply:
And also i don't want to degrade encryption if a mail wasn't parsed successfully. Added a comment on this